pl_bolts/models/detection/components/torchvision_backbones.py

```diff
@@ -0,0 +1,106 @@
+
+import torch.nn as nn
+from typing import Optional
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+try:
+    import torchvision
+
+    from ._supported_models import TORCHVISION_MODEL_ZOO
+except ModuleNotFoundError:
+    warn_missing_pkg('torchvision')  # pragma: no-cover
+
+
+__all__ = ["create_torchvision_backbone"]
+
+
+def _create_backbone_generic(model, out_channels: int):
+    """
+    Generic Backbone creater. It removes the last linear layer.
+
+    Args:
+        model: torch.nn model
+        out_channels: Number of out_channels in last layer.
+    """
+    modules_total = list(model.children())
+    modules = modules_total[:-1]
+    ft_backbone = nn.Sequential(*modules)
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+# Use this when you have Adaptive Pooling layer in End.
+# When Model.features is not applicable.
+def _create_backbone_adaptive(model, out_channels: Optional[int] = None):
+    """
+    Creates backbone by removing linear after Adaptive Pooling layer.
+
+    Args:
+        model: torch.nn model with adaptive pooling layer.
+        out_channels: Number of out_channels in last layer.
+    """
+    if out_channels is None:
+        modules_total = list(model.children())
+        out_channels = modules_total[-1].in_features
+    return _create_backbone_generic(model, out_channels=out_channels)
+
+
+def _create_backbone_features(model, out_channels: int):
+    """
+    Creates backbone from feature sequential block.
+
+    Args:
+        model: torch.nn model with features as sequential block.
+        out_channels: Number of out_channels in last layer.
+    """
+    ft_backbone = model.features
+    ft_backbone.out_channels = out_channels
+    return ft_backbone
+
+
+def create_torchvision_backbone(model_name: str, pretrained: bool = True):
+    """
+    Creates CNN backbone from Torchvision.
+
+    Args:
+        model_name: Name of the model. E.g. resnet18
+        pretrained: Pretrained weights dataset "imagenet", etc
+    """
+
+    if model_name == "mobilenet_v2":
+        model_selected = TORCHVISION_MODEL_ZOO[model_name]
+        net = model_selected(pretrained=pretrained)
+        out_channels = 1280
```

Comment:
`out_channels` sound strange, what is the meaning of channels?

Reply:
`out_channels` is output channels or lastchannel of last layer before `AdaptivePooling`